### DISCUSSION TITLE
Systematics setup (negative and 0 yields, symmetrize down variation to up)

### DISCUSSION THREAD

**Discussion by Tomoya-Iizawa** [scikit-hep/cabinetry#467] [Q&A]
Hi,

I have two questions about systematics setup.
One is that when a bin of nominal distribution has negative yields and up variation has 0, fit fails in the middle.
I will try to set the negative yields to 0, but is this behavior as expected?

Another question is that I think cabinetry does not work if down variation is set and trying to symmetrize it to up, as following,

    Up:
      Symmetrize: True
    Down:
      VariationPath: "JET_JER_EffectiveNP_1__1down"

Is there any way to realize this?

Best,
Tomoya

**Comment by alexander-held** [comment]
> One is that when a bin of nominal distribution has negative yields and up variation has 0, fit fails in the middle.
I will try to set the negative yields to 0, but is this behavior as expected?

Is some specific sample negative in a bin or the full model prediction with all samples stacked? The crash would generally be expected due to the likelihood becoming NaN if the total prediction is negative. One way to help mitigate this are the `clip_sample_data` / `clip_bin_data` settings that `pyhf` provides when creating a model (see https://github.com/scikit-hep/pyhf/pull/1845).

This is not generally a good solution though: such a model is unphysical and should ideally be fixed to not predict such unphysical distributions.

> Another question is that I think cabinetry does not work if down variation is set and trying to symmetrize it to up, as following,

The symmetrization so far only supported `Symmetrize: True` for `Down` variations, but #470 fixes this (I'll also update #385 with these changes). Thanks for raising this!
